### DISCUSSION TITLE
一对一导航如何选择表中多个字段关联

### DISCUSSION THREAD

**Discussion by anhuizf1** [dotnetcore/FreeSql#1073] [General]
比如主表A：treeId,  Id   子表B 也有treeId，Id  ，如何实现一对一多列映射 （select * from A a left join B b on a.treeId=b.treeId and a.Id=b.Id）
代码如下：
public class A {
 public int treeId {get;set;}
  public int Id {get;set;}
   **[Navigate("treeId,Id?")]**
  public B  Son {get;set;}
}

public class B{
 public int treeId {get;set;}
 public int  Id {get;set;}
 public string Name {get;set;}
}

**Comment by 2881099** [comment]
是这样配置的

直接：

```c#
fsql.Select<A>().Include(a => a.B).ToSql()
```

**Reply by anhuizf1** [reply]
不是查，是如何配置多个字段关联

**Reply by 2881099** [reply]
```c#
public class A
{
   [Column(IsPrimary = true)]
   public int treeId {get;set;}
   [Column(IsPrimary = true)]
   public int Id {get;set;}

   [Navigate("treeId,Id")]
   public B Son {get;set;}
}

public class B
{
   [Column(IsPrimary = true)]
   public int treeId {get;set;}
   [Column(IsPrimary = true)]
   public int Id {get;set;}

   public string Name {get;set;}
}
```

**Reply by anhuizf1** [reply]
可以了，Thanks♪(･ω･)ﾉ
